SeqXML parser broken

### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```python
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

CPython
Darwin-19.6.0-x86_64-i386-64bit
1.79.dev0


In `test_SeqIO_SeqXML.py`:

```python
def assert_equal_records(testCase, record_a, record_b):
    testCase.assertEqual(record_a.id, record_b.id)
    testCase.assertEqual(record_a.name, record_b.name)
    testCase.assertEqual(record_a.description, record_b.description)
    testCase.assertEqual(str(record_a.seq), str(record_b.seq))
    testCase.assertEqual(record_a.dbxrefs, record_b.dbxrefs)
    testCase.assertEqual(record_a.annotations, record_a.annotations)
```

The last line compares `record_a` to `record_a` instead of to `record_b`.
Fixing this line causes the test to fail.

Confirmed:

```
======================================================================
FAIL: test_read_write_protein (__main__.TestReadAndWrite)
Read and write protein.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_SeqIO_SeqXML.py", line 180, in test_read_write_protein
    self._write_parse_and_compare(read1_records)
  File "test_SeqIO_SeqXML.py", line 199, in _write_parse_and_compare
    assert_equal_records(self, record1, record2)
  File "test_SeqIO_SeqXML.py", line 22, in assert_equal_records
    testCase.assertEqual(record_a.annotations, record_b.annotations)
AssertionError: {'sou[89 chars]tein', 'sequence status': ['complete'], 'prope[22 chars]one]} != {'sou[89 chars]tein'}
  {'molecule_type': 'protein',
   'ncbi_taxid': '10090',
   'organism': 'Mus musculus',
-  'property with no value': [None],
-  'sequence status': ['complete'],
   'source': 'Ensembl'}

----------------------------------------------------------------------
```

Test case is using ``SeqXML/protein_example.xml`` which does have test lines:

```xml
        <property name="sequence status" value="complete"/>
        <property name="property with no value"/>
```

The test file has not changed in 10 years.

I'm guessing this is an oversight in writing properties, perhaps here:

https://github.com/biopython/biopython/blob/biopython-178/Bio/SeqIO/SeqXmlIO.py#L641

Thomas Schmitt does not seem to be on GitHub anymore...

--- a/Bio/SeqIO/SeqXmlIO.py
+++ b/Bio/SeqIO/SeqXmlIO.py
@@ -653,12 +653,14 @@ class SeqXmlWriter(SequenceWriter):
                 elif isinstance(value, list):
 
                     for v in value:
-                        if isinstance(value, (int, float, str)):
-                            attr = {"name": key, "value": v}
-                            self.xml_generator.startElement(
-                                "property", AttributesImpl(attr)
-                            )
-                            self.xml_generator.endElement("property")
+                        if v is None:
+                            attr = {"name": key}
+                        else:
+                            attr = {"name": key, "value": str(v)}
+                        self.xml_generator.startElement(
+                            "property", AttributesImpl(attr)
+                        )
+                        self.xml_generator.endElement("property")
 
                 elif isinstance(value, (int, float, str)):
 

--- a/Tests/test_SeqIO_SeqXML.py
+++ b/Tests/test_SeqIO_SeqXML.py
@@ -194,7 +194,6 @@ class TestReadAndWrite(unittest.TestCase):
             self.assertEqual(record1.dbxrefs, record2.dbxrefs)
             self.assertEqual(record1.annotations, record2.annotations)
 
-
     def test_write_species(self):
         """Test writing species from annotation tags."""
         record = SeqIO.read("SwissProt/sp016", "swiss")

--- a/Tests/test_SeqIO_SeqXML.py
+++ b/Tests/test_SeqIO_SeqXML.py
@@ -13,15 +13,6 @@ from Bio.SeqRecord import SeqRecord
 from io import BytesIO
 
 
-def assert_equal_records(testCase, record_a, record_b):
-    testCase.assertEqual(record_a.id, record_b.id)
-    testCase.assertEqual(record_a.name, record_b.name)
-    testCase.assertEqual(record_a.description, record_b.description)
-    testCase.assertEqual(str(record_a.seq), str(record_b.seq))
-    testCase.assertEqual(record_a.dbxrefs, record_b.dbxrefs)
-    testCase.assertEqual(record_a.annotations, record_a.annotations)
-
-
 class TestSimpleRead(unittest.TestCase):
     def test_check_SeqIO(self):
         """Files readable using parser via SeqIO."""
@@ -196,7 +187,13 @@ class TestReadAndWrite(unittest.TestCase):
         self.assertEqual(len(read1_records), len(read2_records))
 
         for record1, record2 in zip(read1_records, read2_records):
-            assert_equal_records(self, record1, record2)
+            self.assertEqual(record1.id, record2.id)
+            self.assertEqual(record1.name, record2.name)
+            self.assertEqual(record1.description, record2.description)
+            self.assertEqual(record1.seq, record2.seq)
+            self.assertEqual(record1.dbxrefs, record2.dbxrefs)
+            self.assertEqual(record1.annotations, record2.annotations)
+
 
     def test_write_species(self):
         """Test writing species from annotation tags."""

